plasmapy/utils/decorators/converter.py

```diff
@@ -0,0 +1,47 @@
+"""
+Decorator to convert units of functions in /physics methods
+"""
+__all__ = ["angular_freq_to_hz"]
+
+from astropy import units as u
+import functools
+import inspect
+from plasmapy.utils.decorators import preserve_signature
+
+
+
+def angular_freq_to_hz(fn):
+    """Decorator to convert angular frequencies from radians per second
+     to  Hz.
+
+    Parameters
+    ----------
+    fn: callable
+        The function being wrapped.
+
+    Returns
+    -------
+    callable
+        Wrapped version of the function.
+    -------
+    """
+    wrapped_sign = inspect.signature(fn)
+    fname = fn.__name__
+    @preserve_signature
+    @functools.wraps(fn)
+    def wrapper(*args, **kwargs):
```

Comment:
As `wrapper()` is written, this will cause an `Exception` to be raised if the wrapped function does not defined keyword `to_hz` and/or `**kwargs`.  In my opinion, requiring the wrapped function to define these keywords defeats the purpose of creating a decorator. @namurphy and @StanczakDominik , you can correct me on this if you think differently.  This can easily be corrected by one of two ways...

``` python
def wrapper(*args, to_hz=False, **kwargs):
    
    _result = fn(*args, **kwargs)

    if to_hz:
        return _result.to(u.Hz, equivalencies=[(u.cy/u.s, u.Hz)])
    return _result
```
or
``` python
def wrapper(*args, **kwargs):

    to_hz = kwargs.pop('to_hz', False)

    _result = fn(*args, **kwargs)

    if to_hz:
        return _result.to(u.Hz, equivalencies=[(u.cy/u.s, u.Hz)])
    return _result
```

There's no fundamental difference between the two, but the former is a little more readable and concise.

* You can delete all the `bound_args`, `given_params`, etc.  They were only needed in the decorators you saw because those decorators actually examined the function parameters to determine what it was suppose to do. You don't need that here.

* I'll add another comment to the main PR thread about adding documentation to the wrapped function.  This goes along with the statement I made at the beginning of this comment in reference to not requiring the wrapped function to be code knowing it's decorated.

Reply:
That's a very good point! In fact, now that I think about it, I think something along these lines  was what we had in mind a while discussing - apologies for not recalling that sooner.

I think I like the second option more: the first one is going to fail for functions without `**kwargs` in the signature.

Reply:
The first option would not cause an error since `to_hz` is not in the `kwargs` dictionary, so it's never passed to the wrapped function.  The second option would have that error if `'to_hz'` was not explicitly popped form the `kwargs` dictionary.

Reply:
I'll add, we should probably raise an exception if the wrapped function has defined the `to_hz` keyword.

Reply:
You are, of course, absolutely right, while I am definitely way past my optimal bedtime right now :zzz:. I like the first one now.